Mathlib/Tactic/Multiplicativity.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2023 Arend Mellendijk. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arend Mellendijk
+-/
+
+import Mathlib.Tactic.Multiplicativity.Init
+
+/-!
+# Multiplicativity
+
+We define the multiplicativity tactic using aesop
+
+-/
+
+namespace Nat.ArithmeticFunction
+
+/--
+The `multiplicativity` attribute used to tag `IsMultiplicative` statements for the
+`multiplicativity` tactic. -/
+macro "multiplicativity" : attr =>
+  `(attr|aesop safe apply (rule_sets [$(Lean.mkIdent `IsMultiplicative):ident]))
+
+/--
+`multiplicativity` solves goals of the form `IsMultiplicative f` for `f : ArithmeticFunction R`
+by applying lemmas tagged with the user attribute `multiplicativity`. -/
+macro (name := multiplicativity) "multiplicativity" c:Aesop.tactic_clause* : tactic =>
```

Comment:
The name sounds quite general compared to the actual scope: "multiplicative" has lots of uses like "multiplicative actions", and the `Multiplicative` type synonym. So maybe a more complicated name like `arith_multiplicative`?

Reply:
`fun_prop` is designed for proposition for which you have `*_id`, `*_const`, `*_comp`, `*_apply` and `*_pi` theorems(or subset of them). I do not see them for  `IsMultiplicative` so I don't think it is applicable in this case.

Reply:
Sorry for the late reply - since this was still tagged `awaiting-author` I was expecting you might have more to say/do here.

I see why `fun_prop` wouldn't be appropriate for this tactic then. So the design looks good and I just have a question about the name.

Note that the tactic syntax `multiplicative` is not namespaced:
```
example : Nat.ArithmeticFunction.IsMultiplicative Nat.ArithmeticFunction.moebius := by multiplicativity
```
works without opening any namespace. So I really do think that a more specific name should be used.

Reply:
Sorry for the delay on my part. I agree with changing the name. How about `arith_mult`? `multiplicativity` is frankly a bit of a mouthful.

And I also think `aesop` is probably a little suboptimal regarding performance and code extraction, but for a tactic that won't be very widely used, it should be fine. (that is to say I am pretty excited about `fun_prop`, and I'm a little sad it can't be used here)

There's also a merge conflict at the moment, so I'll merge master and rename.